backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java

```diff
@@ -225,49 +225,62 @@ private Response filterInputOutput(Session session, Response response) {
     return builder.build();
   }
 
-  private Iterable<? extends LineageEntryBatchResponse> filter(Session session,
-      List<LineageEntryBatchResponse> lineageEntryBatchResponses) {
+  private Iterable<? extends LineageEntryBatchResponse> filter(
+      Session session, List<LineageEntryBatchResponse> lineageEntryBatchResponses) {
     final Set<String> experimentRuns = new HashSet<>();
     final Map<Long, RepositoryContainer> blobs = new HashMap<>();
-    return lineageEntryBatchResponses.stream().flatMap(lineageEntryBatchResponse -> {
-      List<LineageEntryBatchResponse> newList = new LinkedList<>();
-      List<LineageEntryBatchResponseSingle> lineageEntryBatchResponseItemsList = lineageEntryBatchResponse
-          .getItemsList();
-      List<LineageEntryBatchResponseSingle> result = lineageEntryBatchResponseItemsList.stream()
-          .flatMap(lineageEntryBatchResponseSingle -> {
-            List<LineageEntryBatchResponseSingle> newLineageEntryBatchResponseSingleList = new LinkedList<>();
-            List<LineageEntry> itemList = lineageEntryBatchResponseSingle.getItemsList();
-            List<LineageEntry> filterResult = itemList.stream().filter(lineageEntry -> {
-              try {
-                validate(session, experimentRuns, blobs, lineageEntry);
-                return true;
-              } catch (StatusRuntimeException | ModelDBException e) {
-                LOGGER.warn("Can't access entity {}", e.getMessage());
-                return false;
-              } catch (NoSuchAlgorithmException | InvalidProtocolBufferException e) {
-                LOGGER.error("Unexpected error {}", e.getMessage());
-                Status status =
-                    Status.newBuilder()
-                        .setCode(com.google.rpc.Code.INTERNAL_VALUE)
-                        .setMessage(e.getMessage())
-                        .addDetails(Any.pack(LineageEntryBatchResponse.getDefaultInstance()))
-                        .build();
-                throw StatusProto.toStatusRuntimeException(status);
-              }
-            }).collect(Collectors.toList());
-            if (filterResult.size() != 0) {
-              newLineageEntryBatchResponseSingleList
-                  .add(LineageEntryBatchResponseSingle.newBuilder()
-                      .setId(lineageEntryBatchResponseSingle.getId()).addAllItems(filterResult)
-                      .build());
-            }
-            return newLineageEntryBatchResponseSingleList.stream();
-          }).collect(Collectors.toList());
-      if (result.size() != 0) {
-        newList.add(LineageEntryBatchResponse.newBuilder().addAllItems(result).build());
-      }
-      return newList.stream();
-    }).collect(Collectors.toList());
+    return lineageEntryBatchResponses.stream()
```

Comment:
This code is considerably harder for me to understand the the previous version that didn't use streams. If you're going to break down into functions, the least you could do is break down into actual functions with some reasonable names so that other people can understand. Could you transform both lambdas from `map` and `flatMap` into their own properly-named functions so that we can understand more easily?

Reply:
fixed